File: 12/solution.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    long long n;
    cin >> n;
    const int nrt = sqrt(n);
    for (int i = 2; i <= nrt + 1; i++)
    {
        int cnt = 0;
        while (n > 0 && n % i == 0)
        {
            n /= i;
            cnt++;
        }
        if (cnt != 0)
        {
            cout << i << " " << cnt << endl;
        }
    }
    if (n != 1)
    {
        cout << n << " " << 1 << endl;
    }
    return 0;
}

File: 12/Solution.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] argS) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int nrt = 1 + (int) Math.sqrt(n);
        for (int i = 2; i < nrt; i++) {
            int cnt = 0;
            while (n > 0 && n % i == 0) {
                n /= i;
                cnt++;
            }

            if (cnt != 0) {
                System.out.printf("%d %d\n", i, cnt);
            }
        }
        if (n != 1) {
            System.out.printf("%d 1\n", n);
        }
    }
}
